src/reflect/scala/reflect/internal/tpe/GlbLubs.scala

```diff
@@ -230,14 +230,17 @@ private[internal] trait GlbLubs {
   def weakLub(tps: List[Type]): Type = (
     if (tps.isEmpty)
       NothingTpe
-    else if (tps forall isNumericValueType)
+    else if (shouldUseNumericLub(tps))
       numericLub(tps)
     else if (tps exists typeHasAnnotations)
       annotationsLub(lub(tps map (_.withoutAnnotations)), tps)
     else
       lub(tps)
   )
 
+  private def shouldUseNumericLub(tps: List[Type]) =
+    tps forall (tp => isNumericValueType(tp.widen))
```

Comment:
A wise man once told me "there ain't no such thing as a small change". I'd like to see how things play out over in the singleton type literals PR.

re: rebasing and retargeting, I'm not depending on this for anything other than excising a single, safe `.asInstanceOf` call. Only say the word and it shall be done.